sky/backends/backend_utils.py

```diff
@@ -798,7 +799,12 @@ def write_cluster_config(
     excluded_clouds = []
     remote_identity = skypilot_config.get_nested(
         (str(cloud).lower(), 'remote_identity'), 'LOCAL_CREDENTIALS')
-    if remote_identity == 'SERVICE_ACCOUNT':
+    if remote_identity is not None and not isinstance(remote_identity, str):
+        for profile in remote_identity:
+            if fnmatch.fnmatchcase(cluster_name_on_cloud, profile):
```

Comment:
Would be nice to match `cluster_name` instead `cluster_name_on_cloud` as the former is user-facing and should be more intuitive to stay the same as the one specified in `~/.sky/config.yaml`.
```suggestion
            if fnmatch.fnmatchcase(cluster_name, profile):
```

Reply:
fixed!